sdk/servicebus/azure-servicebus/stress/scripts/test_stress_queues.py

```diff
@@ -0,0 +1,332 @@
+    sb_client = ServiceBusClient.from_connection_string(
+        SERVICE_BUS_CONNECTION_STR, logging_enable=LOGGING_ENABLE)
+    stress_test = RestartHandlerStressTestRunner(
+                                    senders = [sb_client.get_queue_sender(SERVICEBUS_QUEUE_NAME)],
+                                    receivers = [sb_client.get_queue_receiver(SERVICEBUS_QUEUE_NAME)],
+                                    duration = timedelta(seconds=300),
+                                    receive_delay = 30,
+                                    send_delay = 10,
+                                    azure_monitor_metric=AzureMonitorMetric("test_stress_queue_close_and_reopen")
+                                    )
+
+    result = stress_test.run()
+    assert(result.total_sent > 0)
+    assert(result.total_received > 0)
+
+# This test validates that all individual messages are received contiguously over a long time period.
+# (e.g. not dropped for whatever reason, not sent, or not received)
+class DroppedMessageCheckerStressTestRunner(StressTestRunner):
+    def on_receive(self, state, received_message, receiver):
+        '''Called on every successful receive'''
+        last_seen = getattr(state, 'last_seen', -1)
+        noncontiguous = getattr(state, 'noncontiguous', set())
+        body = int(str(received_message))
+        if body == last_seen+1:
+            last_seen += 1
+            if noncontiguous:
+                while (last_seen+1) in noncontiguous:
+                    last_seen += 1
+                    noncontiguous.remove(last_seen)
+        else:
+            noncontiguous.add(body)
+        state.noncontiguous = noncontiguous
+        state.last_seen = last_seen
+
+    def pre_process_message_body(self, payload):
+        '''Called when constructing message body'''
+        try:
+            body = self._message_id
+        except:
+            _message_id = 0
+            body = 0
+        _message_id += 1
+
+        return str(body)
+
+@pytest.mark.liveTest
+@pytest.mark.live_test_only
+def test_stress_queue_check_for_dropped_messages():
+    sb_client = ServiceBusClient.from_connection_string(
+        SERVICE_BUS_CONNECTION_STR, logging_enable=LOGGING_ENABLE)
+    stress_test = DroppedMessageCheckerStressTestRunner(
+                                    senders = [sb_client.get_queue_sender(SERVICEBUS_QUEUE_NAME)],
+                                    receivers = [sb_client.get_queue_receiver(SERVICEBUS_QUEUE_NAME)],
+                                    receive_type=ReceiveType.pull,
+                                    duration=timedelta(seconds=3000),
+                                    azure_monitor_metric=AzureMonitorMetric("test_stress_queue_check_for_dropped_messages")
+                                    )
+
+    result = stress_test.run()
+    assert(result.total_sent > 0)
+    assert(result.total_received > 0)
+
+if __name__ == '__main__':
+    #parser = ArgumentParser()
+    pytest.main()
```

Comment:
emmm, curious on where this "main" is defined?
what would happen with `python test_stress_queues.py`, would each test method be called?

Reply:
yeah this is one of the ways to run the tests rather than running the `pytest` command. All of the tests are run as normal when `python test_stress_queues.py` is called. We can also pass in all the same flags.

I did it this way b/c I thought we might want to add the ArgmentParser in the future when we have infinite send/receive to take args for stress tests.